internal/schemas/document.go

```diff
@@ -29,14 +30,18 @@ func ResolveDocument(ctx context.Context, d workflow.Document, outputLocation *s
 		}
 
 		location := workflow.GetTempDir()
-		if outputLocation != nil {
-			location = *outputLocation
-		}
 		documentOut, err := registry.ResolveSpeakeasyRegistryBundle(ctx, d, location)
 		if err != nil {
 			return "", err
 		}
 
+		if outputLocation != nil {
+			// Copy actual document out of bundle over to outputLocation
```

Comment:
Maybe just make a note that registry bundles which utilize references wouldn't work here anyways